src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -70,14 +69,27 @@ object RuleUtils {
       }
     }
 
-    def isHybridScanCandidate(entry: IndexLogEntry, inputSourceFiles: Seq[FileInfo]): Boolean = {
+    val hybridScanEnabled = HyperspaceConf.hybridScanEnabled(spark)
+    val hybridScanDeleteEnabled = HyperspaceConf.hybridScanDeleteEnabled(spark)
+    val isDeltaLakeSource = DeltaLakeRuleUtils.isDeltaLakeSource(plan)
+    val sourceRel = getLogicalRelation(plan).get
+
+    def isHybridScanCandidate(index: IndexLogEntry, inputSourceFiles: Seq[FileInfo]): Boolean = {
       // TODO: Some threshold about the similarity of source data files - number of common files or
       //  total size of common files.
       //  See https://github.com/microsoft/hyperspace/issues/159
       // TODO: As in [[PlanSignatureProvider]], Source plan signature comparison is required to
       //  support arbitrary source plans at index creation.
       //  See https://github.com/microsoft/hyperspace/issues/158
 
+      val entry = if (isDeltaLakeSource) {
```

Comment:
@imback82 WDYT about this function? Should we also define new API inside source provider for this?
#270 ref for the design.

Reply:
Yes, we should define new API inside source provider. I think the current approach always requires the Delta jar available.